packages/lib/modules/pool/actions/create/steps/details/useReClammCurrentPrice.ts

```diff
@@ -0,0 +1,35 @@
+import { useTokens } from '@repo/lib/modules/tokens/TokensProvider'
+import { usePoolCreationForm } from '../../PoolCreationFormProvider'
+import { bn } from '@repo/lib/shared/utils/numbers'
+import { useRateProvider } from '../tokens/useRateProvider'
+import { formatUnits } from 'viem'
+
+/**
+ * @returns the current price of the pool in terms of underlying tokens
+ */
+export function useReClammCurrentPrice() {
+  const { poolCreationForm } = usePoolCreationForm()
+  const { poolTokens, network } = poolCreationForm.watch()
+  const { usdValueForTokenAddress } = useTokens()
+
+  const tokenA = poolTokens[0]
+  const tokenB = poolTokens[1]
+
+  const priceTokenA = +usdValueForTokenAddress(tokenA?.address || '', network, '1')
+  const priceTokenB = +usdValueForTokenAddress(tokenB?.address || '', network, '1')
+
+  const { rate: rawRateTokenA } = useRateProvider(tokenA?.rateProvider)
+  const { rate: rawRateTokenB } = useRateProvider(tokenB?.rateProvider)
+
+  // if token has no rate provider, the rate is simply '1'
+  const rateTokenA = rawRateTokenA ? formatUnits(rawRateTokenA, 18) : '1'
+  const rateTokenB = rawRateTokenB ? formatUnits(rawRateTokenB, 18) : '1'
+
+  // always scale wrapped token usd price to underlying token usd price
+  const adjustedPriceTokenA = bn(priceTokenA).div(rateTokenA)
+  const adjustedPriceTokenB = bn(priceTokenB).div(rateTokenB)
+
+  const currentPrice = adjustedPriceTokenA.div(adjustedPriceTokenB)
+
+  return currentPrice
```

Comment:
for creation UI, current price of reclamm must always be in terms of underlying token price